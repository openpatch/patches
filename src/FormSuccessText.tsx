import { ReactNode } from "react";
import { css } from "@emotion/react";
import { Text } from "./Text";
import { Check } from "./icons/shade";
import { Icon } from "./Icon";
import { Box } from "./Box";

export type FormSuccessTextProps = {
  children?: ReactNode;
};

export const FormSuccessText = ({ children }: FormSuccessTextProps) => {
  return (
    <Box
      display="flex"
      alignItems="center"
      css={(theme) => css`
        svg {
          margin-right: ${theme.space.xxsmall};
        }
      `}
    >
      <Icon color="success" size="small">
        <Check />
      </Icon>
      <Text textColor="success.700" role="alert">
        {children}
      </Text>
    </Box>
  );
};
