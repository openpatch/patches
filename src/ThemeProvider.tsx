import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
