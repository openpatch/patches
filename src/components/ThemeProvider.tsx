import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
