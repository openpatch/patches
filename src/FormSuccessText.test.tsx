import { render } from "@testing-library/react";
import { FormSuccessText, FormSuccessTextProps } from "./FormSuccessText";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: FormSuccessTextProps) =>
  render(
    <ThemeProvider>
      <FormSuccessText {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
