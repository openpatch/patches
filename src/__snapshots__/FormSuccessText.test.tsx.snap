// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  scroll-behavior: smooth;
  line-height: 1.5;
  -webkit-text-size-adjust: 100%;
  color: #1F2933;
  letter-spacing: 0;
  font-size: 16px;
  font-family: Rubik,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",sans-serif;
}

.emotion-0 h1 {
  font-size: 32px;
  margin: 0.67em 0;
}

.emotion-0 h1,
.emotion-0 h2,
.emotion-0 h3,
.emotion-0 h4,
.emotion-0 h5,
.emotion-0 h6 {
  font-family: Rubik,system-ui;
}

.emotion-0 hr {
  box-sizing: content-box;
  height: 0;
  overflow: visible;
}

.emotion-0 pre {
  font-family: "Ubuntu Mono",Menlo,monospace;
  font-size: 1em;
}

.emotion-0 a {
  background-color: transparent;
}

.emotion-0 abbr[title] {
  border-bottom: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration: underline dotted;
  text-decoration: underline dotted;
}

.emotion-0 b,
.emotion-0 strong {
  font-weight: bolder;
}

.emotion-0 code,
.emotion-0 kbd,
.emotion-0 samp {
  font-family: "Ubuntu Mono",Menlo,monospace;
  font-size: 1em;
}

.emotion-0 small {
  font-size: 80%;
}

.emotion-0 sub,
.emotion-0 sup {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
}

.emotion-0 sub {
  bottom: -0.25em;
}

.emotion-0 sup {
  top: -0.5em;
}

.emotion-0 img {
  max-width: 100%;
  border-style: none;
}

.emotion-0 button,
.emotion-0 input,
.emotion-0 optgroup,
.emotion-0 select,
.emotion-0 textarea {
  font-family: inherit;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
}

.emotion-0 button,
.emotion-0 input {
  overflow: visible;
}

.emotion-0 button,
.emotion-0 select {
  text-transform: none;
}

.emotion-0 button,
.emotion-0 [type="button"],
.emotion-0 [type="reset"],
.emotion-0 [type="submit"] {
  -webkit-appearance: button;
}

.emotion-0 button::-moz-focus-inner,
.emotion-0 [type="button"]::-moz-focus-inner,
.emotion-0 [type="reset"]::-moz-focus-inner,
.emotion-0 [type="submit"]::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.emotion-0 button:-moz-focusring,
.emotion-0 [type="button"]:-moz-focusring,
.emotion-0 [type="reset"]:-moz-focusring,
.emotion-0 [type="submit"]:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.emotion-0 fieldset {
  padding: 0.35em 0.75em 0.625em;
}

.emotion-0 legend {
  box-sizing: border-box;
  color: inherit;
  display: table;
  max-width: 100%;
  padding: 0;
  white-space: normal;
}

.emotion-0 progress {
  vertical-align: baseline;
}

.emotion-0 textarea {
  overflow: auto;
}

.emotion-0 [type="checkbox"],
.emotion-0 [type="radio"] {
  box-sizing: border-box;
  padding: 0;
}

.emotion-0 [type="number"]::-webkit-inner-spin-button,
.emotion-0 [type="number"]::-webkit-outer-spin-button {
  height: auto;
}

.emotion-0 [type="search"] {
  -webkit-appearance: textfield;
  outline-offset: -2px;
}

.emotion-0 [type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
}

.emotion-0::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.emotion-0 details {
  display: block;
}

.emotion-0 summary {
  display: -webkit-box;
  display: -webkit-list-item;
  display: -ms-list-itembox;
  display: list-item;
}

.emotion-0 template {
  display: none;
}

.emotion-0 [hidden] {
  display: none;
}

.emotion-1 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 svg {
  margin-right: 4px;
}

.emotion-2 {
  position: relative;
  border-radius: 9999px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: #199473;
  width: 24px;
  height: 24px;
}

.emotion-2 svg {
  width: 24px;
  height: 24px;
}

.emotion-2 svg .primary {
  fill: #65D6AD;
}

.emotion-2 svg .secondary {
  fill: #199473;
}

.emotion-3 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  padding: 0px;
  margin: 0px;
  color: #147D64;
}

<div>
  <div
    class="emotion-0"
    id="patches-root"
  >
    <div
      class="emotion-1"
      display="flex"
    >
      <span
        class="emotion-2"
      >
        <svg
          class="icon-check"
          viewBox="0 0 24 24"
        >
          <circle
            class="primary"
            cx="12"
            cy="12"
            r="10"
          />
          <path
            class="secondary"
            d="M10 14.59l6.3-6.3a1 1 0 011.4 1.42l-7 7a1 1 0 01-1.4 0l-3-3a1 1 0 011.4-1.42l2.3 2.3z"
          />
        </svg>
      </span>
      <p
        class="emotion-3"
        role="alert"
      />
    </div>
  </div>
</div>
`;
